loopy/statistics.py

```diff
@@ -0,0 +1,270 @@
+from __future__ import division
+from __future__ import absolute_import
+import six
+
+__copyright__ = "Copyright (C) 2015 James Stevens"
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import loopy as lp
+import warnings
+from islpy import dim_type
+import islpy._isl as isl
+from pymbolic.mapper import CombineMapper
+
+
+class TypedPolyDict:
```

Comment:
I would perhaps be a little more concrete with the class name--like `TypeToOpCountMap` or some such. Otherwise people scratch their head and wonder what on earth this function just returned. That is, unless you have another use for it in mind.